exporter/exporter.go

```diff
@@ -57,8 +59,106 @@ resource "lightstep_metric_dashboard" "exported_dashboard" {
 }
 `
 
-func escapeSpanQuery(input string) string {
-	return strings.Replace(input, "\"", "\\\"", -1)
+const unifiedDashboardTemplate = `
+resource "lightstep_dashboard" "exported_dashboard" {
+  project_name = var.project
+  dashboard_name = "{{.Attributes.Name}}"
+{{range .Attributes.Charts}}
+  chart {
+    name = "{{.Title}}"
+    rank = "{{.Rank}}"
+    type = "{{.ChartType}}"
+{{range .MetricQueries}}
+    query {
+      query_name          = "{{.Name}}"
+      display             = "{{.Display}}"
+      hidden              = {{.Hidden}}
+      query_string        = {{escapeQueryString .TQLQuery}}
+	}
+{{end}}
+  }
+{{end}}
+}
+`
+
+func escapeHCLString(input string) string {
+	input = strings.Replace(input, "\"", "\\\"", -1)
+	input = strings.Replace(input, "\n", "\\\t", -1)
+	input = strings.Replace(input, "\r", "\\\t", -1)
+	input = strings.Replace(input, "\t", "\\\t", -1)
+	input = strings.Replace(input, "\\", "\\\\", -1)
+	return input
+}
+
+func escapeQueryString(input string) string {
+	// Use "heredoc" syntax if the query contains any newlines or other characters that'd
+	// need to be escaped and make the single line representation less convenient to work with.
+	if strings.Contains(input, "\"") ||
+		strings.Contains(input, "\n") ||
+		strings.Contains(input, "\r") ||
+		strings.Contains(input, "\t") ||
+		strings.Contains(input, "\\") {
+		return "<<EOT\n" + input + "\nEOT"
+	} else {
+		return escapeHCLString(input)
+	}
+}
+
+// dashboardUsesQueryString returns true if any chart in the dashboard
+// uses the query string format.
+//
+// The Provider does not support mixing legacy query charts and
+// query-string charts in the same dashboard and will return an error
+// if there is a mix.
```

Comment:
is there a reason we couldnt just return the legacy format (`lightstep_metric_dashboard`) in this case? it will use the `tql` field name for the query strings, but that seems better than just failing

Reply:
Yes, it would work.  My assumption was (1) we're trying to get folks off the legacy format and (2) in our internal usage, there are no such mixed dashboards.  I thought pushing for "please migrate" would be better than perpetuating the deprecated `tql` field.  I don't feel strongly about this though.

Reply:
i think we have some of this internally... e.g. the [machine dashboards](https://app-meta.lightstep.com/lightstep-public/dashboard/alertevaluator-machine-terraform/wZh3WZpp) mix [legacy](https://github.com/lightstep/lightstep/blob/main/terraform/lightstep/lightstep/dashboards_machine.tf#L27-L32) and [query string](https://github.com/lightstep/lightstep/blob/main/terraform/lightstep/lightstep/dashboards_machine.tf#L314-L321). i'm mostly just worried that, as it stands today, "please migrate" means they have to have all their queries stored in editor mode, which may not be the experience they want. as a user, id probably just say "dang, guess i can't use the exporter" rather than "ok, ill change all my queries"

Reply:
Ah, I didn't realize we had mixed dashboards already.  In that case, yes, this should use the legacy format...I'll update the PR.